homeassistant/components/honeywell/climate.py

```diff
@@ -482,31 +487,33 @@ async def async_turn_aux_heat_off(self) -> None:
 
     async def async_update(self) -> None:
         """Get the latest state from the service."""
-        try:
-            await self._device.refresh()
-            self._attr_available = True
-            self._retry = 0
 
-        except UnauthorizedError:
+        async def _login() -> None:
             try:
                 await self._data.client.login()
                 await self._device.refresh()
                 self._attr_available = True
                 self._retry = 0
 
             except (
-                SomeComfortError,
+                AuthError,
                 ClientConnectionError,
                 asyncio.TimeoutError,
             ):
                 self._retry += 1
-                if self._retry > RETRY:
-                    self._attr_available = False
+                self._attr_available = self._retry <= RETRY
+
+        try:
+            await self._device.refresh()
+            self._attr_available = True
+            self._retry = 0
```

Comment:
Same comment as above.

Reply:
I thought I learned this already :) It just reads so much nicer here.